### ISSUE TITLE
Lots of RUF100 false positive for noqa: E501

### ISSUE THREAD

**Issue by LefterisJP** [charliermarsh/ruff#1284]
I upgraded to `ruff==0.0.186` and started getting lots of `RUF100 Unused `noqa` directive for: E501` errors.

But they are all wrong. There is different kinds of categories but what I see is (assume 99 line length)

#### Directive at end of comment

```python
# this is a veryyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy long comment  # noqa: E501
```

#### Directive at end of multiline docstring
```python
    """This is an adjustment of web3's event data decoding to work with our code
    source: https://github.com/ethereum/web3.py/blob/ffe59daf10edc19ee5f05227b25bac8d090e8aa4/web3/_utils/events.py#L201

    Returns a tuple containing the decoded topic data and decoded log data.

    May raise:
    - DeserializationError if the abi string is invalid or abi or log topics/data do not match
    """  # noqa: E501
```

#### Directive at end of multiline string

Same as above, just for `"""` strings that are not function docstrings

**Comment by charliermarsh** [comment]
Thanks, will take a look at this tonight.

**Comment by charliermarsh** [comment]
(The lines aren’t getting flagged as E501 though, right?)

**Comment by charliermarsh** [comment]
Sorry, bad/lazy application of De Morgan's Law.

**Comment by LefterisJP** [comment]
Heh that was fast. Thank you
